pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -453,6 +472,22 @@ SocketAddress serverAddrees() {
         return future;
     }
 
+    public CompletableFuture<MessageIdData> sendGetLastMessageId(ByteBuf request, long requestId) {
+        CompletableFuture<MessageIdData> future = new CompletableFuture<>();
+
+        pendingGetLastMessageIdRequests.put(requestId, future);
```

Comment:
@merlimat how are timeouts handled with Pulsar's RPC?

Reply:
Unfortunately we don't have a general way of handling it.